File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionLong.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;

public class TransformerFunctionLong extends TransformerFunction {
    public TransformerFunctionLong() {
        super();
    }
    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        return context.getLong(null);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionLong.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";

class TransformerFunctionLong extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    return context.getLong(null);
  }
}

export default TransformerFunctionLong;
